src/dmd/parse.d

```diff
@@ -8483,6 +8483,14 @@ final class Parser(AST) : Lexer
         auto e = parseCondExp();
         while (1)
         {
+            // require parens for e.g. `t ? a = 1 : b = 2`
+            // Deprecated in 2018-05.
+            // Change to error in 2019-05.
+            // @@@DEPRECATED_2019-05@@@.
+            if (e.op == TOK.question && !e.parens && precedence[token.value] == PREC.assign)
+                dmd.errors.deprecation(e.loc, "`%s` must be parenthesized when next to operator `%s`",
+                    e.toChars(), Token.toChars(token.value));
```

Comment:
I suggest that the deprecation message contains a suggestion on how to add the parentheses to get the same behavior as before the deprecation was added. The changelog should reflect this as well.

Reply:
> assuming the reader knows what parenthesized means

I had to refer to the changelog example to understand what it meant.

Reply:
Would "surrounded by parentheses" be better?

Reply:
@andralex and @WalterBright aren't showing any interest in this, so I'm ready to move forward without their approval, but could you please consider @jacob-carlborg's suggestion with regard to the error message?

Reply:
> Edit: Made the change myself.

Ah I think we pushed in the same moment. Anyhow, I force-pushed my variant as the tests need updating too. Sorry.